R/pmcmc_varied_parameter.R

```diff
@@ -0,0 +1,66 @@
+pmcmc_varied_parameter <- function(name, initial, min = -Inf, max = Inf,
+                            discrete = FALSE, prior = NULL) {
+  assert_scalar_character(name)
+  assert_scalar_logical(discrete)
+  assert_is(min, c("numeric", "integer"))
+  assert_is(max, c("numeric", "integer"))
+
+  lng <- length(initial)
+
+  if (length(min) == 1) {
+    min <- rep(min, lng)
+  } else if (length(min) != lng) {
+    stop(sprintf("Length of 'min' must be '1' or %s.", lng))
+  }
+
+  if (length(max) == 1) {
+    max <- rep(max, lng)
+  } else if (length(max) != lng) {
+    stop(sprintf("Length of 'max' must be '1' or %s.", lng))
+  }
+
+  if (any(initial < min)) {
+    stop(sprintf("'initial' must be >= 'min' (%s)", min))
+  }
+  if (any(initial > max)) {
+    stop(sprintf("'initial' must be <= 'max' (%s)", max))
+  }
+
+  if (is.null(prior)) {
+    prior <- rep(list(function(p) 0), lng)
+  } else {
+    if (is.function(prior)) {
+      prior <- rep(list(prior), lng)
+    } else {
+      assert_is(prior, "list")
+      if (length(prior) == 1) {
+        prior <- rep(prior, lng)
+      } else if (length(prior) != lng) {
+        stop(sprintf("Length of 'prior' must be '1' or %s.", lng))
+      }
+      if (any(!vlapply(prior, inherits, what = "function"))) {
+        stop("'prior' should be a list of functions.")
+      }
+    }
+
+    mapply(function(x, y) {
```

Comment:
This is awkward to read, but not sure on this for now

Reply:
Could change to something like (pseudo):

```R
err <- mapply(x(y), {
    value <- try(x(y), silent = TRUE)
    if (class(value)[1] != "try-error") is.finite(value) else FALSE
})
if (any(err)) stop(infinite or not evaluated)
```